komodo/build.py

```diff
@@ -162,58 +176,79 @@ def make(pkgfile, repofile, data,
     seen = set()
     pkgorder = []
     for x in xs:
-        if x in seen: continue
+        if x in seen:
+            continue
         seen.add(x)
         pkgorder.append(x)
 
     fakeprefix = fakeroot + prefix
-    shell(['mkdir -p', fakeprefix])
+    shell(["mkdir -p", fakeprefix])
     prefix = os.path.abspath(prefix)
 
     # assuming there always is a python *and* that python will be installed
     # before pip is required. This dependency *must* be explicit in the
     # repository
-    os.environ['DESTDIR'] = fakeroot
-    os.environ['BOOST_ROOT'] = fakeprefix
-    build_ld_lib_path = ':'.join(filter(None,
-                                            [os.path.join(fakeprefix, 'lib'),
-                                            os.path.join(fakeprefix, 'lib64'),
-                                            os.environ.get('LD_LIBRARY_PATH')]))
-    extra_makeopts = os.environ.get('extra_makeopts')
-    build_pythonpath = pypaths(fakeprefix, pkgs.get('python'))
-    build_path = ':'.join([os.path.join(fakeprefix, 'bin'), os.environ['PATH']])
-
-    pkgpaths = ['{}-{}'.format(pkg, pkgs[pkg]) for pkg in pkgorder]
+    os.environ["DESTDIR"] = fakeroot
+    os.environ["BOOST_ROOT"] = fakeprefix
+    build_ld_lib_path = ":".join(
+        filter(
+            None,
+            [
+                os.path.join(fakeprefix, "lib"),
+                os.path.join(fakeprefix, "lib64"),
+                os.environ.get("LD_LIBRARY_PATH"),
+            ],
+        )
+    )
+    extra_makeopts = os.environ.get("extra_makeopts")
+    build_pythonpath = pypaths(fakeprefix, pkgs.get("python"))
+    build_path = ":".join([os.path.join(fakeprefix, "bin"), os.environ["PATH"]])
+
+    pkgpaths = ["{}-{}".format(pkg, pkgs[pkg]) for pkg in pkgorder]
     if dlprefix:
         pkgpaths = [os.path.join(dlprefix, path) for path in pkgpaths]
 
     def resolve(x):
-        return x.replace('$(prefix)', prefix)
+        return x.replace("$(prefix)", prefix)
 
-    build = { 'rpm': rpm, 'cmake': cmake, 'sh': sh, 'pip': pip_install, 'rsync': rsync, 'noop': noop }
+    build = {
+        "rpm": rpm,
+        "cmake": cmake,
+        "sh": sh,
+        "pip": pip_install,
+        "rsync": rsync,
+        "noop": noop,
+    }
 
     for pkg, path in zip(pkgorder, pkgpaths):
         ver = pkgs[pkg]
         current = repo[pkg][ver]
-        make = current['make']
+        make = current["make"]
         pkgpath = os.path.abspath(path)
+        package_name = pkgs.get("package_name", None)
```

Comment:
I think we should only support `package_name` for `source: pypi`, so do an assertion here.

Basically, if `package_name`, the `source` must be `pypi`.

Reply:
Added